internal/controllers/configuration/kongupstreampolicy_utils.go

```diff
@@ -106,6 +110,51 @@ func (r *KongUpstreamPolicyReconciler) getServicesReferencingUpstreamPolicy(
 	return services.Items, nil
 }
 
+// getIngressesReferencingService fetches all Ingresses using a Service as its backend.
+func (r *KongUpstreamPolicyReconciler) getIngressesReferencingService(
+	ctx context.Context,
+	serviceNN k8stypes.NamespacedName,
+) ([]netv1.Ingress, error) {
+	ingresses := &netv1.IngressList{}
+	if err := r.List(ctx, ingresses,
+		client.InNamespace(serviceNN.Namespace),
+		client.MatchingFields{routeBackendRefServiceNameIndexKey: serviceNN.String()},
+	); err != nil {
+		return nil, fmt.Errorf("failed listing ingresses: %w", err)
+	}
+	return ingresses.Items, nil
+}
+
+// getHTTPRoutesReferencingService fetches all HTTPRoutes using a service as its backend.
+func (r *KongUpstreamPolicyReconciler) getHTTPRoutesReferencingService(
+	ctx context.Context,
+	serviceNN k8stypes.NamespacedName,
+) ([]gatewayapi.HTTPRoute, error) {
+	httpRoutes := &gatewayapi.HTTPRouteList{}
+	if err := r.List(ctx, httpRoutes,
+		client.InNamespace(serviceNN.Namespace),
```

Comment:
An HTTPRoute may refer to a service from a different namespace.

Reply:
Fixed.